waspc/src/Analyzer/Parser/Lexer.x

```diff
@@ -119,21 +126,27 @@ lexer parseToken = do
 -- | Takes a lexeme like "{=json" and sets the quoter start code
 beginQuoter :: String -> Parser Token
 beginQuoter leftQuoteTag = do
-  setStartCode quoter
   let tag = drop 2 leftQuoteTag
+  setStartCode $ QuoterStartCode tag
   createConstToken (TLQuote tag) leftQuoteTag
 
--- | Takes a lexeme like "json=}" and returns to start code 0
-endQuoter :: String -> Parser Token
-endQuoter rightQuoteTag = do
-  setStartCode 0
-  let tag = take (length rightQuoteTag - 2) rightQuoteTag
-  createConstToken (TRQuote tag) rightQuoteTag
+-- | Takes a lexeme like "json=}" and either ends a quoter or add quoted text to the quoter
+lexQuoterEndTag :: String -> Parser Token
+lexQuoterEndTag rightQuoteTag = gets parserLexerStartCode >>= \startCode -> case startCode of
+  DefaultStartCode -> do
```

Comment:
I had a comment here, but I probably deleted it accidentally. That case will never happen (since `lexQuoterEndTag` is only used when the start code is `QuoterStartCode`), so my options were to do this or do `DefaultStartCode -> error "impossible: lexQuoterEndTag with DefaultStartCode"` (or something to that effect). Which sounds better?

Reply:
I would prefer `error` in that case, as it indicates to whoever is reading the code that this should  never happen, and if it somehow does, we are warned about it.